### PR TITLE
Add tokens to drivers.

### PR DESCRIPTION
**Pull request by shans**

### CODE REVIEW

File: src/runtime/storageNG/drivers/volatile.ts [hunk 1]
```diff
@@ -45,6 +45,7 @@ export class VolatileStorageKey extends StorageKey {
 
 export class VolatileMemory {
   entries = new Map<string, VolatileEntry<unknown>>();
+  token = Math.random() + '';
```
Comment (csilvestrini) [inline src/runtime/storageNG/drivers/volatile.ts]: How come the token is a random number here? This isn't immediately obvious to me. (Answering in the form of a comment would be ideal :wink: )

File: src/runtime/storageNG/drivers/driver-factory.ts [hunk 2]
```diff
@@ -37,9 +37,16 @@ export abstract class Driver<Data> {
     this.storageKey = storageKey;
     this.exists = exists;
   }
-  abstract registerReceiver(receiver: ReceiveMethod<Data>): void;
+  abstract registerReceiver(receiver: ReceiveMethod<Data>, token?: string): void;
   abstract async send(model: Data, version: number): Promise<boolean>;
 
+  // Return a token that represents the current state of the data.
+  // This can be provided to registerReceiver, and will impact what
```
Comment (jasonwyatt) [inline src/runtime/storageNG/drivers/driver-factory.ts]: Does this mean that registering multiple receivers will mean each subsequent call can clobber the token from the previous call?

Comment (jasonwyatt) [reply src/runtime/storageNG/drivers/driver-factory.ts]: Oh, nevermind. Looking at volatile - it seems like only one receiver is allowed to be registered at a time.  Maybe this should be `setReceiver`?

Comment (shans) [reply src/runtime/storageNG/drivers/driver-factory.ts]: Yeah, I think you're right.

**Pull request merged**